src/instruction/instructions/load/ldr.ts

```diff
@@ -164,60 +232,53 @@ export class LdrRegisterInstruction extends BaseInstruction {
     registers.writeRegister(
       getBits(opcode[0], this.rtPattern).value,
       memory.readWord(
-        registers
-          .readRegister(Register.PC)
-          .add(getBits(opcode[0], this.immPattern).value)
+        Word.fromUnsignedInteger(
+          //TODO  VCB-176 --> LdrRegisterInstruction.alignPointerToNextWord(registers.readRegister(Register.PC).value)
+          registers.readRegister(Register.PC).value +
+            getImmediateBits(opcode[0], this.immPattern, 0).value //TODO VCB-176 --> getImmediateBits 2 instead of 0
+        )
       )
     )
   }
-}
-
-/**
- * Represents a 'LOAD' instruction - LDR (pointer + offset) - word
- */
-export class LdrLabelInstruction extends BaseInstruction {
-  public name: string = 'LDR'
-  public pattern: string = '01001XXXXXXXXXXX'
-  private immPattern: string = '01001000XXXXXXXX'
-  private rtPattern: string = '01001XXX00000000'
-  private expectedOptionCount: number = 2
 
-  public canEncodeInstruction(name: string, options: string[]): boolean {
+  /**
+   * Determines whether the specified instruction is a
+   * pseudo instruction.
+   *
+   * @param options parameter provided for instruction
+   * @returns whether the instruction is a pseudo instruction
+   */
+  public static isLabelOffsetInstruction(options: string[]): boolean {
     return (
-      super.canEncodeInstruction(name, options) &&
-      isOptionCountValid(options, this.expectedOptionCount)
+      options.length === 2 && LdrRegisterInstruction.isLiteralString(options[1])
     )
   }
 
-  public encodeInstruction(
-    options: string[],
-    labels?: ILabelOffsets
-  ): Halfword[] {
-    checkOptionCount(options, this.expectedOptionCount)
-    let opcode: Halfword = create(this.pattern)
-    opcode = setBits(opcode, this.rtPattern, createLowRegisterBits(options[0]))
-    opcode = setBits(
-      opcode,
-      this.immPattern,
-      labels
-        ? Halfword.fromUnsignedInteger(labels[options[1]].value)
-        : Halfword.fromUnsignedInteger(0x0)
-    )
-    return [opcode]
+  /**
+   * Determines whether the specified string is a
+   * a literal (all except valid register, and strings containing any brackets)
+   *
+   * @param option string provided as param which could be literal
+   * @returns whether the string is a literal part of pseudo instruction
+   */
+  public static isLiteralString(option: string): boolean {
+    try {
+      $enum(Register).getValueOrThrow(option)
+      return false
+    } catch (e) {
+      return !option.includes('[') && !option.includes(']')
+    }
   }
 
```

Comment:
you are right static makes no sense here. added isLiteralString to opcode and isLabelOffsetInstruction as function to ldr file since it is used by all classes within ldr file, but should only be used there as far as I know
also moved pointer alignment () to opcode

Reply:
Very nice, thanks!